internal/data/player_options.go

```diff
@@ -1,35 +1,34 @@
 package data
 
-import "github.com/jinzhu/gorm"
+import (
+	"errors"
+	"gorm.io/gorm"
+)
 
 type PlayerOptions struct {
 	gorm.Model
 
+	Account   *Account
 	AccountID int
-	Account   Account
 
 	KeyConfig []byte
 }
 
 // FindPlayerOptions returns all of hte PlayerOptions associated with an Account.
 func FindPlayerOptions(account *Account) (*PlayerOptions, error) {
 	var playerOptions PlayerOptions
-	q := db.Model(&account).Related(&playerOptions)
+	err := db.First(&playerOptions).Where("account_id = ?", &account.ID).Error
 
-	if q.Error != nil {
-		if gorm.IsRecordNotFoundError(q.Error) {
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
 			return nil, nil
 		}
-		return nil, q.Error
+		return nil, err
 	}
 
 	return &playerOptions, nil
 }
 
-// UpdatePlayerOptions updates the PlayerOptions row with the contents in po.
-func UpdatePlayerOptions(po *PlayerOptions) error {
```

Comment:
This method probably should have been called `UpsertPlayerOptions`. Regardless I'd prefer to keep the semantics of this method for now, the change you've made here leaves us without a way to update them

Reply:
Can I just add another method for update instead? It's not used now but if it needed it looks cleaner to have two different ones for me.

Reply:
Yeah that's fine, my longer term idea was to always just create-or-update player options but given they're guaranteed to at least exist due to the login flow it's reasonable to have separate methods

Reply:
Done